setuptools/monkey.py

```diff
@@ -152,13 +153,13 @@ def patch_for_msvc_specialized_compiler():
     Patch functions in distutils to use standalone Microsoft Visual C++
     compilers.
     """
-    # import late to avoid circular imports on Python < 3.5
-    msvc = import_module('setuptools.msvc')
-
-    if platform.system() != 'Windows':
+    if not get_platform().startswith('win'):
```

Comment:
This fix doesn't feel right to me. It's using a non-standard method to determine the platform. Why is `platform.system()` not suitable? If using a `startswith` comparator, I think a separate function should be supplied with a docstring explaining what's being checked here.